pkg/shim/scheduler.go

```diff
@@ -152,20 +160,42 @@ func (ss *KubernetesShim) recoverSchedulerState() func(e *fsm.Event) {
 		// do not block main thread
 		go func() {
 			log.Logger.Info("recovering scheduler states")
-			// wait for recovery, max timeout 3 minutes
-			if err := ss.context.WaitForRecovery(3 * time.Minute); err != nil {
+			// step 1: recover all applications
+			// this step, we collect all the existing allocated pods from api-server,
+			// identify the scheduling identity (aka applicationInfo) from the pod,
+			// and then add these applications to the scheduler.
+			if err := ss.appManager.WaitForRecovery(3 * time.Minute); err != nil {
 				// failed
 				log.Logger.Fatal("scheduler recovery failed", zap.Error(err))
 				dispatcher.Dispatch(ShimSchedulerEvent{
 					event: events.RecoverSchedulerFailed,
 				})
-			} else {
-				// success
-				log.Logger.Info("scheduler recovery succeed")
+			}
+
+			// step 2: recover existing allocations
+			// this step, we collect all existing allocations (allocated pods) from api-server,
+			// rerun the scheduling for these allocations in order to restore scheduler-state,
+			// the rerun is like a replay, not a actual scheduling procedure.
+			recoverableApps := make([]interfaces.Recoverable, 0)
```

Comment:
`recoverableApps` is not a list of apps it is a list of managers.
When reviewing I was put on the wrong track by the naming: it should be called `mgrs`

This also flows on into the parameter naming of the function `context.WaitForRecovery`, `context.getExistingAllocation` and for `context.recover` it should be `mgrs` not _mgr_

Reply:
fixed